addons/binding/org.openhab.binding.systeminfo/src/main/java/org/openhab/binding/systeminfo/internal/model/OshiSysteminfo.java

```diff
@@ -237,27 +262,39 @@ public DecimalType getStorageUsedPercent(int deviceIndex) throws DeviceNotFoundE
 
     @Override
     public StringType getStorageName(int index) throws DeviceNotFoundException {
+        // In the current OSHI version a new query is required for the storage data values to be updated
+        // In OSHI 4.0.0. it is planed to change this mechanism - see https://github.com/oshi/oshi/issues/310
+        fileStores = operatingSystem.getFileSystem().getFileStores();
```

Comment:
I have taught about this, better remove this line. We can optimize the performance here a little bit to avoid multiple queries by the OSHI library. 
For fields that do not change dynamically, I would avoid a new query (the storage name will most probably never change :), and if it still changes it will updated by calls from the other methods ).